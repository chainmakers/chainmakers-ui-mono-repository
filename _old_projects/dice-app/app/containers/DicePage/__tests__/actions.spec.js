// @flow

import {
  startKmdiceBetting,
  insertKmdiceBetting,
  startKmdiceBettingSuccess,
  startKmdiceBettingError,
  syncKmdiceBetting
} from '../actions';
import {
  KMDICE_BETTING_START,
  KMDICE_BETTING_INSERT,
  KMDICE_BETTING_START_SUCCESS,
  KMDICE_BETTING_START_ERROR,
  KMDICE_BETTING_SYNC
} from '../constants';
import type { StartKmdiceBettingSuccessPayload, ErrorRPCType } from '../schema';

const time = new Date('1995-12-17T03:24:00');

describe('containers/DicePage/actions/startKmdiceBetting', () => {
  const startKmdiceBettingPayload = {
    numberToBet: 23,
    amount: 0.1
  };

  it('should startKmdiceBetting should create startKmdiceBetting action', () => {
    expect(startKmdiceBetting(startKmdiceBettingPayload)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: KMDICE_BETTING_START,
      payload: startKmdiceBettingPayload
    };

    expect(startKmdiceBetting(startKmdiceBettingPayload)).toEqual(
      expectedResult
    );
  });
});

describe('containers/DicePage/actions/insertKmdiceBetting', () => {
  const insertKmdiceBettingPayload = {
    id: 1,
    time,
    numberToBet: 23,
    amount: 0.1
  };

  it('should insertKmdiceBetting should create insertKmdiceBetting action', () => {
    expect(insertKmdiceBetting(insertKmdiceBettingPayload)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: KMDICE_BETTING_INSERT,
      payload: insertKmdiceBettingPayload
    };

    expect(insertKmdiceBetting(insertKmdiceBettingPayload)).toEqual(
      expectedResult
    );
  });
});

describe('containers/DicePage/actions/startKmdiceBettingSuccess', () => {
  const startKmdiceBettingSuccessPayload: StartKmdiceBettingSuccessPayload = {
    bettxid: 'bettxid',
    id: 1,
    status: 'PENDING'
  };

  it('should startKmdiceBettingSuccess should create startKmdiceBettingSuccess action', () => {
    expect(
      startKmdiceBettingSuccess(startKmdiceBettingSuccessPayload)
    ).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: KMDICE_BETTING_START_SUCCESS,
      payload: startKmdiceBettingSuccessPayload
    };

    expect(startKmdiceBettingSuccess(startKmdiceBettingSuccessPayload)).toEqual(
      expectedResult
    );
  });
});

describe('containers/DicePage/actions/startKmdiceBettingError', () => {
  const errorRPCType: ErrorRPCType = {
    context: {
      action: 'rpc:call',
      params: []
    },
    type: 'komodod',
    message: 'no message',
    ok: 'failed'
  };

  it('should startKmdiceBettingError should create startKmdiceBettingError action', () => {
    expect(startKmdiceBettingError(errorRPCType)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: KMDICE_BETTING_START_ERROR,
      payload: errorRPCType
    };

    expect(startKmdiceBettingError(errorRPCType)).toEqual(expectedResult);
  });
});

describe('containers/DicePage/actions/syncKmdiceBetting', () => {
  const syncKmdiceBettingPayload: SyncKmdiceBettingPayload = {
    bettxid: 'bettxid',
    id: 1
  };

  it('should syncKmdiceBetting should create syncKmdiceBetting action', () => {
    expect(syncKmdiceBetting(syncKmdiceBettingPayload)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: KMDICE_BETTING_SYNC,
      payload: syncKmdiceBettingPayload
    };

    expect(syncKmdiceBetting(syncKmdiceBettingPayload)).toEqual(expectedResult);
  });
});
