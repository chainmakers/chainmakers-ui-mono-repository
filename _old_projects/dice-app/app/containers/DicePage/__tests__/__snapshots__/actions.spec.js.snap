// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/DicePage/actions/insertKmdiceBetting should insertKmdiceBetting should create insertKmdiceBetting action 1`] = `
Object {
  "payload": Object {
    "amount": 0.1,
    "id": 1,
    "numberToBet": 23,
    "time": 1995-12-16T20:24:00.000Z,
  },
  "type": "kmdice/DicePage/KMDICE_BETTING_INSERT",
}
`;

exports[`containers/DicePage/actions/startKmdiceBetting should startKmdiceBetting should create startKmdiceBetting action 1`] = `
Object {
  "payload": Object {
    "amount": 0.1,
    "numberToBet": 23,
  },
  "type": "kmdice/DicePage/KMDICE_BETTING_START",
}
`;

exports[`containers/DicePage/actions/startKmdiceBettingError should startKmdiceBettingError should create startKmdiceBettingError action 1`] = `
Object {
  "payload": Object {
    "context": Object {
      "action": "rpc:call",
      "params": Array [],
    },
    "message": "no message",
    "ok": "failed",
    "type": "komodod",
  },
  "type": "kmdice/DicePage/KMDICE_BETTING_START_ERROR",
}
`;

exports[`containers/DicePage/actions/startKmdiceBettingSuccess should startKmdiceBettingSuccess should create startKmdiceBettingSuccess action 1`] = `
Object {
  "payload": Object {
    "bettxid": "bettxid",
    "id": 1,
    "status": "PENDING",
  },
  "type": "kmdice/DicePage/KMDICE_BETTING_START_SUCCESS",
}
`;

exports[`containers/DicePage/actions/syncKmdiceBetting should syncKmdiceBetting should create syncKmdiceBetting action 1`] = `
Object {
  "payload": Object {
    "bettxid": "bettxid",
    "id": 1,
  },
  "type": "kmdice/DicePage/KMDICE_BETTING_SYNC",
}
`;
